emcc.py

```diff
@@ -1217,7 +1217,10 @@ def check(input_file):
         newargs.append('-D__EMSCRIPTEN_PTHREADS__=1')
         shared.Settings.FORCE_FILESYSTEM = 1 # proxying of utime requires the filesystem
       else:
-        js_libraries.append(shared.path_from_root('src', 'library_pthread_stub.js'))
+        if final_suffix in JS_CONTAINING_SUFFIXES:
```

Comment:
I'd prefer using C files over conversion to asm.js, since it means they can potentially be reused for the wasm backend.